src/OpenApi/gen/XmlCommentGenerator.Parser.cs

```diff
@@ -0,0 +1,172 @@
+            if (name is not null)
+            {
+                comments.Add((name, member.ToString()));
+            }
+        }
+        return comments;
+    }
+
+    internal static List<(string, string)> ParseCompilation(Compilation compilation, CancellationToken cancellationToken)
+    {
+        var visitor = new AssemblyTypeSymbolsVisitor(compilation.Assembly, cancellationToken);
+        visitor.VisitAssembly();
+        var types = visitor.GetPublicTypes();
+        var comments = new List<(string, string)>();
+        foreach (var type in types)
+        {
+            if (DocumentationCommentId.CreateDeclarationId(type) is string name &&
+                type.GetDocumentationCommentXml(CultureInfo.InvariantCulture, expandIncludes: true, cancellationToken: cancellationToken) is string xml)
+            {
+                comments.Add((name, xml));
+            }
+        }
+        var properties = visitor.GetPublicProperties();
+        foreach (var property in properties)
+        {
+            if (DocumentationCommentId.CreateDeclarationId(property) is string name &&
+                property.GetDocumentationCommentXml(CultureInfo.InvariantCulture, expandIncludes: true, cancellationToken: cancellationToken) is string xml)
+            {
+                comments.Add((name, xml));
+            }
+        }
+        var methods = visitor.GetPublicMethods();
+        foreach (var method in methods)
+        {
+            // If the method is a constructor for a record, skip it because we will have already processed the type.
+            if (method.MethodKind == MethodKind.Constructor)
+            {
+                continue;
+            }
+            if (DocumentationCommentId.CreateDeclarationId(method) is string name &&
+                method.GetDocumentationCommentXml(CultureInfo.InvariantCulture, expandIncludes: true, cancellationToken: cancellationToken) is string xml)
+            {
+                comments.Add((name, xml));
+            }
+        }
+        return comments;
+    }
+
+    // Type names are used in a `typeof()` expression, so we need to replace generic arguments
+    // with empty strings to avoid compiler errors.
+    private static string ReplaceGenericArguments(string typeName)
+    {
+        var stack = new Stack<int>();
+        var result = new StringBuilder(typeName);
+
+        for (var i = 0; i < result.Length; i++)
+        {
+            if (result[i] == '<')
+            {
+                stack.Push(i);
+            }
+            else if (result[i] == '>' && stack.Count > 0)
+            {
+                var start = stack.Pop();
+                // Replace everything between < and > with empty strings separated by commas
+                var segment = result.ToString(start + 1, i - start - 1);
+                var commaCount = segment.Count(c => c == ',');
+                var replacement = string.Join(",", Enumerable.Repeat("", commaCount + 1));
```

Comment:
This would be equivalent I think if I'm reading this correctly.

```suggestion
                var replacement = new string(',', commaCount + 1);
```

Reply:
Good catch! I thinj for the delta we'll want to do just `commaCount` instead of `commaCount + 1`.